### DISCUSSION TITLE
NEED HELP: How to get original documents from Faiss index?

### DISCUSSION THREAD

**Discussion by cuichenxu** [castorini/pyserini#1939] [Q&A]
```
from pyserini.search import FaissSearcher

searcher = FaissSearcher(
    '.../indexs',   # index dir
    '..../DPR'  # DPR dir
)
hits = searcher.search('what is a Ed file.')

for i in range(0, len(hits)):
    print(f'{i+1:2} {hits[i].docid:7} {hits[i].score:.5f}')

```
The code is as above, Does Pyserini support to get original document from Faiss index?
Or is there any ways to get original document of retrieved docid?

**Comment by lintool** [comment]
Hi @cuichenxu - your question is answered here: https://github.com/castorini/pyserini/discussions/1593
